File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
def solution(array):
    count = [0] * (max(array) + 1)

    for i in array:
        count[i] += 1

    max_count = max(count)

    if count.count(max_count) > 1:
        return -1

    return count.index(max_count)

File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.cs
using System;

public class Solution {
    public int solution(int[] array) {
        int max = 0;

        foreach (int num in array) {
            if (num > max) {
                max = num;
            }
        }

        int[] count = new int[max + 1];

        foreach (int num in array) {
            count[num]++;
        }

        int maxCount = 0;

        foreach (int num in count) {
            if (num > maxCount) {
                maxCount = num;
            }
        }

        int answer = -1;
        int same = 0;

        for (int i = 0; i < count.Length; i++) {
            if (count[i] == maxCount) {
                answer = i;
                same++;
            }
        }

        if (same > 1) {
            return -1;
        }

        return answer;
    }
}
